### PR TITLE
Remove dependency on named memory mapped files

### PR DESCRIPTION
**Pull request by megakid**
Just a small one for starters, looking at building an integration into Obvs (https://github.com/christopherread/Obvs) so we can start to utilize this nice lib.

Solves #72 
Removed the named memory mapped file usage.

### CODE REVIEW

**Comment by UncleFirefox** [comment]
Hi! Thanks for your contribution! Nice to know that other initiatives join us! @JPWatson what do you think? Should we merge and bump the current version up to stable?

**Comment by megakid** [comment]
I have added a further commit to send in `null` rather than `Guid.NewGuid().ToString()` as the map name.  This, I think, will enable Linux running of the library.

File: src/Adaptive.Agrona.Tests/AtomicBufferTests.cs [hunk 1]
```diff
@@ -38,14 +41,58 @@ public class AtomicBufferTests
...
+        private IAtomicBuffer AlignedByteArrayBacked;
+
+        private IAtomicBuffer UnmanagedBacked;
 
-        private static readonly MappedByteBuffer MappedByteBuffer = new MappedByteBuffer(MemoryMappedFile.CreateNew("testmap", BufferCapacity));
+        private IAtomicBuffer MemoryMappedFileBacked;
+        private bool Initialized;
 
-        [Datapoint] public static readonly IAtomicBuffer MemoryMappedFileBacked = new UnsafeBuffer(MappedByteBuffer.Pointer, BufferCapacity);
+        public void Initialize()
```
Comment (JPWatson) [inline src/Adaptive.Agrona.Tests/AtomicBufferTests.cs]: clearer to have a [OneTimeSetup] and remove the Initialized flag and call from Datapoints()?

File: src/Adaptive.Agrona/IoUtil.cs [hunk 2]
```diff
@@ -60,9 +60,9 @@ public static MemoryMappedFile OpenMemoryMappedFile(string path, MapMode mapMode
             var f = new FileStream(path, FileMode.Open, fileAccess, fileShare);
 
 #if NETFULL
-            return MemoryMappedFile.CreateFromFile(f, Guid.NewGuid().ToString(), 0, memoryMappedFileAccess, new MemoryMappedFileSecurity(), HandleInheritability.None, false);
+            return MemoryMappedFile.CreateFromFile(f, null, 0, memoryMappedFileAccess, new MemoryMappedFileSecurity(), HandleInheritability.None, false);
```
Comment (JPWatson) [inline src/Adaptive.Agrona/IoUtil.cs]: Is there any documentation around this behaviour?

File: src/Adaptive.Agrona.Tests/AtomicBufferTests.cs (hunk 1 above)
Comment (megakid) [reply src/Adaptive.Agrona.Tests/AtomicBufferTests.cs]: I thought the same.  Unfortunately (and I tried all kinds of ways to do this), `[OneTimeSetup]` does not run before the `Datapoints` are evaluated.  I tried all kinds of different orders/things and this was the only way that worked.  Happy to re-evaluate if there's another way.

File: src/Adaptive.Agrona/IoUtil.cs (hunk 2 above)
Comment (megakid) [reply src/Adaptive.Agrona/IoUtil.cs]: Unfortunately it seems a bit hazy - my source was https://github.com/dotnet/corefx/issues/20091#issuecomment-303079998

But looking at the documentation here - https://msdn.microsoft.com/en-us/library/dn804422(v=vs.110).aspx - it says `ArgumentException when mapName is null or an empty string.`. 

However, in the official corefx tests, it has `null` being used https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CrossProcess.cs#L13

See here for linux:

https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs#L22-L34

But then the main class has this:

https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs#L202-L205

Maybe this needs reverting.

Comment (JPWatson) [reply src/Adaptive.Agrona/IoUtil.cs]: That last example is fine too - `(mapName != null && mapName.Length == 0)`.
So it's just the documentation that disagrees, happy to trust the code.

File: src/Adaptive.Agrona.Tests/AtomicBufferTests.cs (hunk 1 above)
Comment (JPWatson) [reply src/Adaptive.Agrona.Tests/AtomicBufferTests.cs]: Fair enough - that's not really intuitive behaviour from nunit.

**Comment by JPWatson** [comment]
👍 Thanks @megakid

**Pull request merged**